### PR TITLE
Smarter CSV parsing

### PR DESCRIPTION
**Pull request by jimallman**
This revision uses [Papa Parse](https://www.papaparse.com/) (a mature and more robust CSV/TSV parser) in place of my too-simple DIY parsing.

### CODE REVIEW

**Comment by jimallman** [comment]
Incidentally, this updated bundle should also fix our [current JS vulnerabilities](https://github.com/OpenTreeOfLife/opentree/security/dependabot).

**Comment by snacktavish** [comment]
Parsing looks good, but mapping still seems a bit odd - it appears to follow the text string of the name, rather than the OTT_id.

fro example on study:
https://devtree.opentreeoflife.org/curator/study/edit/tt_266?tab=otu-mapping&match=Acan
 if I paste in Acanthophoenix rubra otuPS1, "Bernieria madagascariensis (Gmelin, 1789)", 959901
(our same troublesome taxon https://tree.opentreeoflife.org/taxonomy/browse?id=959901)

it seems to ignore the ottid and map to the (broken) name itself.

image

**Comment by jimallman** [comment]
Wow. This is happening because of the extra whitespace in your delimited values. [Strictly speaking](https://en.wikipedia.org/wiki/Comma-separated_values#Basic_rules), the delimiter is _just_ a comma, and any following spaces are considered part of the next field value _and must be preserved_. So for your input:
```
Acanthophoenix rubra otuPS1, "Bernieria madagascariensis (Gmelin, 1789)", 959901
```
the fields will be parsed as
```
`Acanthophoenix rubra otuPS1`
` "Bernieria madagascariensis (Gmelin`
` 1789)"`
` 959901`
```
Not only are the unwanted spaces being captured, but in doing so this _breaks_ Papa Parse's normally correct handling of the quoted field with inner comma. 😌 

In short, we might need very sparse CSV input with no extra whitespace. But I'll be there's a configuration option in Papa Parse to handle this. Let's see...

**Comment by jimallman** [comment]
I should have mentioned -- if you try the same input without leading whitespace, the behavior is as expected!
`Acanthophoenix rubra otuPS1,"Bernieria madagascariensis (Gmelin, 1789)",959901`

**Comment by jimallman** [comment]
UPDATE: There's no proper workaround for this in Papa Parse, none that would correctly handle our edge case (quoted field with inner comma and leading whitespace). The Papa Parse maintainers actually [seem](https://github.com/mholt/PapaParse/issues/241) a little [prickly](https://github.com/mholt/PapaParse/issues/330) on this [subject](https://github.com/mholt/PapaParse/issues/731), so we might need to encourage people to make strictly valid CSV, at least if they have trouble with their first submission.

File: curator/static/js/tnrs/tnrs-main.js [hunk 1]
```diff
@@ -116,47 +117,51 @@ var getNewNamesetModel = function(options) {
     */
     return obj;
 };
+
+var papaParseConfig = {
+    /* Try to accomodate different delimiters, line endings, etc.
+     * For all settings and default values, see <https://www.papaparse.com/docs#config>
+     */
+    delimiter: "",	// auto-detect
+    delimitersToGuess: [', ', ',', '\t', '|', ';', Papa.RECORD_SEP, Papa.UNIT_SEP],
```
Comment (jimallman) [inline curator/static/js/tnrs/tnrs-main.js:126]: Sadly, suggesting ', ' as a possible delimiter didn't fix our leading-space problem. It seems to be ignored or rejected by Papa Parse, though [their code seems to accept multi-character delimiters](https://github.com/mholt/PapaParse/blob/e11ee26581d9f497087b4c3ee00c2a77645b81d2/papaparse.js#L1294). Odd.

**Pull request merged**